### PR TITLE
build: add header check to comply with TRG 7.02

### LINKED ISSUE #88: Fix license header to match Eclipse recommendations
Please fix license header in src code files to match Eclipse recommendations.

### PR DESCRIPTION
**Pull request by ndr-brt**
## WHAT

Add a check on checkstyle to ensure the headers are in the correct format according to [TRG 7.02](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02/)

## WHY

TRG compliance

## FURTHER NOTES

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

Closes #88

### CODE REVIEW

**Review by wolf4ood — APPROVED** [review]
:rocket:

File: build.gradle.kts [hunk 1]
```diff
@@ -109,11 +111,18 @@ allprojects {
         configFile = rootProject.file("resources/tx-checkstyle-config.xml")
         configDirectory.set(rootProject.file("resources"))
 
+        tasks.checkstyleMain {
```
Comment (paullatzelsperger) [inline build.gradle.kts:114]: why do we need this? shouldn't `checkstyleMain` run on the `main` source set anyway?

Comment (ndr-brt) [reply build.gradle.kts:114]: I was looking into checking also `resources` file (as `.ServiceExtension` files), but the gradle checkstyle plugin by default is excluding everything not java:
https://github.com/gradle/gradle/blob/f8ee6cae5bd056483bafec72043744540f532e09/platforms/jvm/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java#L139

in any case I forgot to do it :sweat_smile:  will do it right now

Comment (paullatzelsperger) [reply build.gradle.kts:114]: interesting 🤔 maybe a comment would be useful then?

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**